tests/s25Main/UI/testVideoDriver.cpp

```diff
@@ -100,3 +102,121 @@ BOOST_FIXTURE_TEST_CASE(CreateAndDestroyTextures, uiHelper::Fixture)
+        BOOST_TEST(Position(guiScale.viewToScreen(Position(400, 300))) == Position(800, 600));
+    }
+
+    {
+        // screenToView rounds scalars and points
+        driver->setGuiScalePercent(95);
+        const auto& guiScale = driver->getGuiScale();
+        constexpr auto pt = Position(10, 15);
+        const auto pt1 = guiScale.screenToView<Position>(pt);
+        const auto pt2 = Position(guiScale.screenToView<int>(static_cast<float>(pt.x)),
+                                  guiScale.screenToView<int>(static_cast<float>(pt.y)));
+        const auto pt3 = Position(Position::Truncate, PointF(pt) * driver->getGuiScale().invScaleFactor());
+        BOOST_TEST(pt1 == pt2);
+        BOOST_TEST(pt1 != pt3);
+        // Points are off by 1 due to rounding
+        BOOST_TEST((pt1 - pt3) == Position(1, 1));
+    }
+
+    {
+        // viewToScreen rounds scalars and points
+        driver->setGuiScalePercent(105);
+        const auto& guiScale = driver->getGuiScale();
+        constexpr auto pt = Position(10, 15);
+        const auto pt1 = guiScale.viewToScreen<Position>(pt);
+        const auto pt2 = Position(guiScale.viewToScreen<int>(static_cast<float>(pt.x)),
+                                  guiScale.viewToScreen<int>(static_cast<float>(pt.y)));
+        const auto pt3 = Position(Position::Truncate, PointF(pt) * driver->getGuiScale().scaleFactor());
+        BOOST_TEST(pt1 == pt2);
+        BOOST_TEST(pt1 != pt3);
+        // Points are off by 1 due to rounding
+        BOOST_TEST((pt1 - pt3) == Position(1, 1));
+    }
+
+    {
+        // screenToView and viewToScreen round scalars and points
+        driver->setGuiScalePercent(rttr::test::randomValue<unsigned>(90, 110));
+        const auto& guiScale = driver->getGuiScale();
+        const auto pt = rttr::test::randomPoint<Position>(-100, 100);
+        BOOST_TEST(guiScale.screenToView<Position>(pt)
+                   == Position(guiScale.screenToView<int>(static_cast<float>(pt.x)),
+                               guiScale.screenToView<int>(static_cast<float>(pt.y))));
+        BOOST_TEST(guiScale.viewToScreen<Position>(pt)
+                   == Position(guiScale.viewToScreen<int>(static_cast<float>(pt.x)),
+                               guiScale.viewToScreen<int>(static_cast<float>(pt.y))));
+    }
+}
+
+BOOST_AUTO_TEST_CASE(GuiScaleRange)
+{
+    unsigned minPercent, maxPercent, recommendedPercent;
+    auto* driver = uiHelper::GetVideoDriver();
+    driver->CreateScreen("", VideoMode(800 * 3, 600 * 3), false);
+
+    // Regular DPI configuration
+    driver->getGuiScaleRange(&minPercent, &maxPercent, &recommendedPercent);
+    BOOST_TEST(minPercent == 100u);
+    BOOST_TEST(maxPercent == 300u);
+    BOOST_TEST(recommendedPercent == 100u);
+
+    // Maximum is constrained by width
+    driver->ResizeScreen(VideoMode(800 * 2, 600 * 3), false);
+    driver->getGuiScaleRange(nullptr, &maxPercent, nullptr);
+    BOOST_TEST(maxPercent == 200u);
```

Comment:
This could look like:
```suggestion
    BOOST_TEST(driver->getGuiScaleRange().max == 200u);
```

I think that's easier to read/use, isn't it?